.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+repos:
+-   repo: https://github.com/pre-commit/mirrors-autopep8
```

Comment:
Does this end up applying to the whole file when you commit it? Does that make it likely to generate large style diffs?

Reply:
Yes. I would suggest we don't allow anyone to use it until we lint the entire package. But I thought to add it now.

Reply:
hmmm, I wonder if this is useful before then anyway.

Reply:
If you dont have pre commit installed it wont do anything or tell you its here.

Reply:
I was more meaning people could install it and use it before we apply it to all the files in the repo, for isort and the more basic checks.

Reply:
Sure but if we never tell people it exists they wont hopefully?